### DISCUSSION TITLE
How do I close a dialog derived from ttkbootstrap.dialogs.Dialog

### DISCUSSION THREAD

**Discussion by stroudcuster** [israel-dryer/ttkbootstrap#358] [Q&A] [answered]
How do I close a subclass of ttkbootstrap.dialogs.Dialog.  My dialog displays properly and I'm handling a button press event to close the dialog, but so far the .destroy and .quit methods don't remove the dialog from the screen.

class CSVImportStatusDialog(dialogs.Dialog):
    def __init__(self, parent, status: imp.ImportStatus):
        dialogs.Dialog.__init__(self, parent, 'CSV Import Process Status')
        self.status: imp.ImportStatus = status
        self.close_button = None

    def create_body(self, parent):
        if self.status.success:
            ok = 'Successful'
        else:
            ok = 'Errors were encountered.'
        ttkb.Label(parent, text=f'Import Process Status: {ok}').grid(column=0, row=0, padx=5, pady=5)
        msg_count = self.status.msg_count()
        msg_scroll = scrolled.ScrolledText(parent, padding=5, height=msg_count)
        for msg in self.status.messages:
            msg_scroll.insert(END, f'{msg}\n')
        msg_scroll.grid(column=0, row=1, rowspan=msg_count)
        self.close_button = ttkb.Button(parent, text="Close", command=self.close_dialog)
        self.close_button.grid(column=0, row=self.status.msg_count() + 2)

    def create_buttonbox(self, parent):
        pass

    def close_dialog(self):
        self.quit()
        self.update()

**Comment by israel-dryer** [comment, marked as answer]
`Dialog` isn't the window itself — it's a wrapper that builds the actual window as `self._toplevel` when you call `show()`. That's why `self.quit()` (which just exits the mainloop) and `self.destroy()` (which destroys the wrapper, not the visible window) don't close it.

Close the toplevel directly:

```python
def close_dialog(self):
    self._toplevel.destroy()
```

This is the same thing the base class binds `<Escape>` to internally, so it's the intended way. Your `create_buttonbox` -> `pass` plus a custom Close button is fine; just point the button's `command` at `self._toplevel.destroy()`.

Follow-up: reaching into the private `_toplevel` isn't ideal, so I've opened #1155 to add a public `Dialog.close()` method for the 2.0 cleanup. Once that lands, `self.close()` will be the tidy way to do this.
